### PR TITLE
Clean up library a bit

### PR DESCRIPTION
**Pull request by Kelwing**
- Cleans up the library to reduce code reuse
- Add rest of Interactions endpoints
- Fix a few bugs with wrong Method type, and wrong response code
- Add a few other missing endpoints
- Remove a few removed endpoints
- Added thread support

### CODE REVIEW

**Review by ZachyFoxx — COMMENTED** [review]
Other than the one other comment, this LGTM.

File: voice.go [hunk 1]
```diff
@@ -7,22 +7,14 @@ import (
...
-	if err != nil {
-		return nil, err
-	}
+	regions := []*objects.VoiceRegion{}
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(VoiceRegions).
+		ContentType(JsonContentType).
+		Expect(http.StatusOK).
+		Bind(regions).
```
Comment (Jack073) [inline voice.go:16]: We need to pass a pointer to the slice here, otherwise it will automatically return an error.

File: template.go [hunk 2]
```diff
@@ -9,65 +9,41 @@ import (
...
-	}
-
-	return templates, nil
+	templates := []*objects.Template{}
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(fmt.Sprintf(GuildTemplateFmt, guild)).
+		Expect(http.StatusOK).
+		ContentType(JsonContentType).
+		Bind(templates).
```
Comment (Jack073) [inline template.go:44]: We need to pass a pointer to the slice here, otherwise it will automatically return an error.

File: guild.go [hunk 3]
```diff
@@ -137,66 +110,40 @@ func (c *Client) ModifyGuild(id objects.Snowflake, params *ModifyGuildParams) (*
...
-	if err = res.JSON(&channels); err != nil {
-		return nil, err
-	}
+	channels := []*objects.Channel{}
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(fmt.Sprintf(GuildChannelsFmt, id)).
+		ContentType(JsonContentType).
+		Expect(http.StatusOK).
+		Bind(channels).
```
Comment (Jack073) [inline guild.go:143]: We need to pass a pointer to the slice here, otherwise it will automatically return an error.

File: guild.go [hunk 4]
```diff
@@ -258,44 +196,40 @@ func (c *Client) ModifyGuildChannelPositions(id objects.Snowflake, params []*Mod
...
-		return err
-	}
+func (c *Client) ListActiveThreads(id objects.Snowflake) ([]*ListThreadsResponse, error) {
+	channels := []*ListThreadsResponse{}
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(fmt.Sprintf(GuildThreadsFmt, id)).
+		ContentType(JsonContentType).
+		Expect(http.StatusOK).
+		Bind(channels).
```
Comment (Jack073) [inline guild.go:216]: We need to pass a pointer to the slice here, otherwise it will automatically return an error.

File: guild.go [hunk 5]
```diff
@@ -315,25 +249,16 @@ func (c *Client) ListGuildMembers(guild objects.Snowflake, params *ListGuildMemb
...
-	if err != nil {
-		return nil, err
-	}
+	members := []*objects.GuildMember{}
+	err = NewRequest().
+		Method(http.MethodGet).
+		Path(u.String()).
+		ContentType(JsonContentType).
+		Expect(http.StatusOK).
+		Bind(members).
```
Comment (Jack073) [inline guild.go:258]: We need to pass a pointer to the slice here, otherwise it will automatically return an error.

File: guild.go [hunk 6]
```diff
@@ -435,121 +344,74 @@ func (c *Client) ModifyCurrentUserNick(guild objects.Snowflake, params *ModifyCu
...
-	if err != nil {
-		return nil, err
-	}
+	bans := []*objects.Ban{}
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(fmt.Sprintf(GuildBansFmt, guild)).
+		ContentType(JsonContentType).
+		Expect(http.StatusOK).
+		Bind(bans).
```
Comment (Jack073) [inline guild.go:398]: We need to pass a pointer to the slice here, otherwise it will automatically return an error.

File: guild.go [hunk 7]
```diff
@@ -563,61 +425,35 @@ func (c *Client) CreateBan(guild, user objects.Snowflake, params *CreateGuildBan
...
-	}
-
-	return roles, nil
+	roles := []*objects.Role{}
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(fmt.Sprintf(GuildRolesFmt, guild)).
+		ContentType(JsonContentType).
+		Expect(http.StatusOK).
+		Bind(roles).
```
Comment (Jack073) [inline guild.go:454]: We need to pass a pointer to the slice here, otherwise it will automatically return an error.

File: guild.go [hunk 8]
```diff
@@ -673,26 +499,17 @@ func (c *Client) ModifyGuildRolePositions(guild objects.Snowflake, params []*Mod
...
-		return nil, err
-	}
+	roles := []*objects.Role{}
+	err = NewRequest().
+		Method(http.MethodPatch).
+		Path(fmt.Sprintf(GuildRolesFmt, guild)).
+		ContentType(JsonContentType).
+		Body(data).
+		Expect(http.StatusOK).
+		Bind(roles).
```
Comment (Jack073) [inline guild.go:509]: We need to pass a pointer to the slice here, otherwise it will automatically return an error.

**Review by Jack073 — CHANGES_REQUESTED** [review]
It's the same thing 8 times, just that's all the places I saw it and figured it was easier to mark it like this

**Comment by Kelwing** [comment]
> It's the same thing 8 times, just that's all the places I saw it and figured it was easier to mark it like this

I'll fix this once I'm done working today 😄

**Review by Jack073 — APPROVED** [review]
All LGTM

**Review by JakeMakesStuff — APPROVED** [review]
lgtm

**Pull request merged**